test/slidingwindow-test.js

```diff
@@ -0,0 +1,135 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+'use strict';
+
+const assert = require('bsert');
+const base32 = require('bs32');
+
+const rules = require('../lib/covenants/rules');
+const SlidingWindow = require('../lib/net/slidingwindow');
+const FullNode = require('../lib/node/fullnode');
+const packets = require('../lib/net/packets');
+const packetTypes = packets.types;
+
+const common = require('../test/util/common');
+
+async function sleep(time) {
+  return new Promise(resolve => setTimeout(resolve, time));
+}
+
+describe('SlidingWindow (Unit)', function() {
+  const window = new SlidingWindow({
+    window: 10,
+    limit: 100
+  });
+
+  beforeEach(() => {
+    window.start();
+  });
+
+  afterEach(() => {
+    window.stop();
+  });
+
+  it('should process max requests in window', async () => {
+    for (let i=0; i < window.limit-1; i++)
+      window.increase(1);
+
+    assert.ok(window.allow());
+  });
+
+  it('should reject after max requests in window', async () => {
+    window.increase(1);
+    assert.ok(!window.allow());
+  });
+
+  it('should reset after window timeout', async () => {
+    let reset = false;
+
+    window.once('reset', () => {
+      reset = true;
+    });
+
+    await sleep(window.window);
+    assert.ok(reset === true);
+  });
+});
+
+describe('SlidingWindow (Functional)', function() {
+  it('should rate limit getproof to max-proof-rps ', async () => {
+    const maxProofRPS = 20;
+
+    const one = new FullNode({
+      'memory': true,
+      'network': 'regtest',
+      'listen': true,
+      'max-proof-rps': maxProofRPS,
+      'host': '127.0.0.1',
```

Comment:
I don't think we technically need to specify the hosts since now we are using the ports to differentiate.